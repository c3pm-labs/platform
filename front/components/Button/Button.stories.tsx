import { Meta } from '@storybook/react';

import Button, { ButtonProps } from '.';

export default {
  title: 'Button',
  component: Button,
} as Meta;

export const basic = (args: ButtonProps): JSX.Element => (<Button {...args}>Hello</Button>);
