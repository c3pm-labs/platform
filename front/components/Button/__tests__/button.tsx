import { render } from '@testing-library/react';

import Button from '../index';

test('Button', () => {
  const { container } = render(<Button>Hello</Button>);

  expect(container).toMatchSnapshot();
});

test('Button contained', () => {
  const { container } = render(<Button variant="contained">Hello</Button>);

  expect(container).toMatchSnapshot();
});

test('Button outlined', () => {
  const { container } = render(<Button variant="outlined">Hello</Button>);

  expect(container).toMatchSnapshot();
});
