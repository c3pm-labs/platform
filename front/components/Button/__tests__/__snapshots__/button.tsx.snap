// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-default-1 makeStyles-primary-2 makeStyles-primary-6 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button contained 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-10 makeStyles-primary-11 makeStyles-primary-15 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Button outlined 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-default-19 makeStyles-primary-20 makeStyles-primary-24 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Hello
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
